calicoctl/commands/node/run.go

```diff
@@ -256,6 +274,58 @@ Description:
 		fmt.Println(err)
 		os.Exit(1)
 	}
+
+	// Create the command to follow the docker logs for the calico/node
+	logCmd := exec.Command("docker", "logs", "calico-node", "--follow")
+
+	// Get the stdout pipe
+	outPipe, err := logCmd.StdoutPipe()
+	if err != nil {
+		fmt.Println("Error querying calico/node logs:")
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Start following the logs.
+	err = logCmd.Start()
+	if err != nil {
+		fmt.Println("Error querying calico/node logs:")
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Read stdout until the node fails, or until we see the output
+	// indicating success.
+	stdout := ""
+	txt := make([]byte, 1000)
+	var n int
+	for {
```

Comment:
shouldn't we have a timout of some sort here? This loop could go on forever if we simply rely on docker process to fail or finish as docker daemon hangs many times and for those cases this would be in the loop forever